crates/steel-core/src/steel_vm/primitives.rs

```diff
@@ -894,11 +951,17 @@ fn expt(left: &SteelVal, right: &SteelVal) -> Result<SteelVal> {
 #[steel_derive::function(name = "exp", constant = true)]
 fn exp(left: &SteelVal) -> Result<SteelVal> {
     match left {
-        SteelVal::IntV(i) if *i == 0 => Ok(SteelVal::IntV(1)),
+        SteelVal::IntV(0) => Ok(SteelVal::IntV(1)),
```

Comment:
Kind of interesting. Guile gives `1` for this, but Racket gives `1.0`.

Reply:
Well I was definitely trying to replicate racket behavior - I guess also maybe racket tries to convert to exact where possible, but that also might be a completely baseless statement